src/fsharp/IlxGen.fs

```diff
@@ -2072,7 +2072,13 @@ and GenConstant cenv cgbuf eenv (c,m,ty) sequel =
           | Const.SByte i -> CG.EmitInstr cgbuf (pop 0) (Push [ilTy]) (mkLdcInt32 (int32 i))
           | Const.Int16 i -> CG.EmitInstr cgbuf (pop 0) (Push [ilTy]) (mkLdcInt32 (int32 i))
           | Const.Int32 i -> CG.EmitInstr cgbuf (pop 0) (Push [ilTy]) (mkLdcInt32 i)
-          | Const.Int64 i -> CG.EmitInstr cgbuf (pop 0) (Push [ilTy]) (iLdcInt64 i)
+          | Const.Int64 i -> 
+            if i >= int64 System.Int32.MinValue && i <= int64 System.Int32.MaxValue then
```

Comment:
please add a comment here that links to this issue here

Reply:
As this is only a PR, should we make a dedicated issue?

Reply:
for me this is fine since you explained the situation

Reply:
Hm sorry for disturbing but I disagree, we should open an issue for this so people can find  it if necessary in the issues.

Reply:
If the contributors reach consensus I am happy to create another summary for this in an issue ;)